pinot-broker/src/main/java/org/apache/pinot/broker/routing/segmentselector/RealtimeSegmentSelector.java

```diff
@@ -52,8 +53,8 @@
   public static final String FORCE_HLC = "FORCE_HLC";
 
   private final AtomicLong _requestId = new AtomicLong();
-  private volatile List<List<String>> _hlcSegments;
```

Comment:
Is it possible to not have this interface change of List to Set by handling the logic inside the pruner. I believe the change of List to Set comes because otherwise you need a memcpy in the prune() method?

Reply:
Yes, I made this change to avoid making a HashSet of input segments in `prune()`. Is there any considerations to keep this as list? The time range pruner keeps a search treemap from time ranges to segments. Pruning is done by search intersected segments based on query and check if they are in input segments of `prune()`. Ideally the input segments are in a HashSet, so the checking process would be efficient.